code/modules/overmap/interdiction.dm

```diff
@@ -0,0 +1,180 @@
+		if(tgui_alert(user, "Cancel Interdiction", "Interdiction", list("Yes", "No")) != "Yes")
+			return
+		interdicting.interdictor = null
+		interdicting = null
+		say("Released Interdiction.")
+		return
+
+	var/list/targets = scan_for_targets()
+	if(targets == FALSE)
+		return
+	if(!length(targets))
+		say("No valid targets in sensor range.")
+		return
+
+	var/list/sorted_targets = sortList(targets, .proc/cmp_ship_dist)
+	var/list/options = list()
+	for(var/datum/overmap/ship/controlled/target as anything in sorted_targets)
+		options["[get_dist(current_ship.token, target.token)] - [target.name]"] = target
+
+	var/target = tgui_input_list(user, "Select Ship", "Interdiction", options)
+	if(!target || !(target in options))
+		return
+
+	do_interdiction(user, options[target])
+
+/obj/machinery/computer/interdiction/proc/cmp_ship_dist(datum/overmap/t1, datum/overmap/t2)
+	return cmp_numeric_asc(get_dist(current_ship.token, t1.token), get_dist(current_ship.token, t2.token))
+
+/obj/machinery/computer/interdiction/proc/throw_ship_contents(datum/overmap/ship/controlled/target, dir, strength)
+	for(var/area/area as anything in target.shuttle_port.shuttle_areas)
+		for(var/atom/movable/movable in area)
+			var/turf/target_turf = get_edge_target_turf(movable, dir)
+			movable.throw_at(target_turf, 2 * strength, strength)
+
+/obj/machinery/computer/interdiction/proc/do_interdiction(mob/user, datum/overmap/ship/controlled/target)
+	message_admins("[user] is attempting to perform an interdiction to [target] from [current_ship]")
+
+	var/target_x
+	var/target_y
+	if(length(target.get_nearby_overmap_objects()))
+		var/target_found = FALSE
+		for(var/dir in GLOB.cardinals|GLOB.diagonals)
+			if(target_found)
+				break
+			var/c_x = target.x
+			var/c_y = target.y
+			switch(dir)
+				if(NORTH)
+					c_y++
+				if(SOUTH)
+					c_y--
+				if(EAST)
+					c_x++
+				if(WEST)
+					c_x--
+				if(NORTHWEST)
+					c_y++
+					c_x--
+				if(NORTHEAST)
+					c_y++
+					c_x++
+				if(SOUTHWEST)
+					c_y--
+					c_x--
+				if(SOUTHEAST)
+					c_y--
+					c_x++
+			var/list/overmap_square = SSovermap.overmap_container[c_x][c_y]
+			if(length(overmap_square))
+				continue
+			target_x = c_x
+			target_y = c_y
+			target_found = TRUE
+
+		if(!target_found)
+			say("Failed to find valid Interdiction point!")
+			return
+	else
+		target_x = target.x
+		target_y = target.y
+
+	interdicting = target
+	target.interdictor = src
+
+	var/their_speed = target.get_speed()
+	if(their_speed >= 20) // this is gonna hurt, you and me
+		if(tgui_alert(user, "Radars indicitate they are going very quickly. Are you sure you want to do this?", "Interdiction", list("Yes", "No"), 10 SECONDS) != "Yes")
+			interdicting = null
+			target.interdictor = null
+			return
+
+		var/heading = target.get_heading()
+		throw_ship_contents(target, heading, their_speed * SPEED_THROW_STRENGTH_MULT)
+		throw_ship_contents(current_ship, turn(heading, 180), their_speed * SPEED_THROW_STRENGTH_MULT)
```

Comment:
Maybe make this speed differential instead of just raw speed? So if you're both going fast it's easier &c